metaflow/plugins/catch_decorator.py

```diff
@@ -105,14 +111,12 @@ def task_decorate(
 
         # if the user code has failed max_user_code_retries times, @catch
         # runs a piece of fallback code instead. This way we can continue
-        # running the flow downsteam, as we have a proper entry for this task.
+        # running the flow downstream, as we have a proper entry for this task.
 
         def fallback_step(inputs=None):
             raise FailureHandledByCatch(retry_count)
 
-        # We don't run fallback for `ubf_control` since it doesn't support
-        # any retries.
-        if ubf_context != UBF_CONTROL and retry_count > max_user_code_retries:
+        if retry_count > max_user_code_retries:
```

Comment:
Why was this stripped? I get that here the control task is also a mapper but, unless there is a reason to, can we keep the control-and-not-a-mapper mode as well?

Reply:
This is not related to control as mapper, but just enabling catch semantics for the control task. Before, @catch was not supported for UBF control tasks.

Reply:
right. That was by design in a sense since in the control task is not a worker initial paradigm, the control task wasn't written by the user so we wouldn't catch any errors from it. Here, the control is also a mapper so it makes sense for it to follow the catch semantics but in the previous version (which I agree is no longer used but unless there is a strong reason to remove, we could keep it as is) the control task was not user code.

Reply:
@tuulos  do you want to chime in? I think you also suggested that catch should work with UBF as expected.

Reply:
@romain-intel the old code path didn't support `@catch` for UBF which was a deliberate choice at the time for the use cases we considered back then. Now that we are considering different use cases, we need to enable `@catch` for control.

In case we get some other UBF use cases in the future that require different behavior, we can make it configurable by the decorator. This is a reversible change in that sense